library/src/main/java/com/yausername/youtubedl_android/YoutubeDL.java

```diff
@@ -179,6 +178,13 @@ public YoutubeDLResponse execute(YoutubeDLRequest request, @Nullable DownloadPro
 
         try {
             process = processBuilder.start();
+            Pattern pattern = Pattern.compile("(?<=\\[)(?:[^\\[\\]]+|\\[[^\\[\\]]+\\])+");
+            Matcher matcher = pattern.matcher(process.toString());
+            if (matcher.find()) {
+                String pidInfo[] = matcher.group().split(",\\s+");
+                String pidPair[] = pidInfo[0].split("=");
+                pidCallback.onProcessRun(pidPair[1]);
```

Comment:
You need to check pidCallback for nullity. Otherwise java will throw NullPointerException when pidCallback is not specified.